src/mol-util/color/scale.ts

```diff
@@ -100,4 +101,82 @@ export namespace ColorScale {
             get legend() { return ScaleLegend(minLabel, maxLabel, colors); }
         };
     }
+
+    export function createDiscrete(props: ColorScaleProps): ColorScale {
```

Comment:
This has a lot of the same code as `ColorScale.create`. Can you find ways so they share more code?

Reply:
I personally don't mind the copy-paste and small modification. General rule of thumb would be to do this at most 3 times though before refactoring.

Reply:
I have tried my best to make it more generic, check it out now, please.

I have also handled the case when the user enters the custom thresholds in a non-sorted order.